packages/web-components/src/popover/popover.ts

```diff
@@ -0,0 +1,129 @@
+import { attr, FASTElement, Updates } from '@microsoft/fast-element';
+import { arrow, computePosition, offset } from '@floating-ui/dom';
+
+export class Popover extends FASTElement {
+  @attr({ mode: 'boolean' })
+  public open: boolean = false;
+
+  @attr({ mode: 'boolean' })
+  public arrow: boolean = false;
+
+  @attr
+  public position: 'top' | 'bottom' | 'left' | 'right' = 'top';
+
+  public popoverContentRef: FASTElement; // @FIXME: what type?
+  public arrowRef: FASTElement;
+  public anchorRef: HTMLElement[];
+
+  public connectedCallback() {
+    super.connectedCallback();
+    this.openChanged(); // hack
+  }
+
+  public disconnectedCallback() {
+    super.disconnectedCallback();
+    this.hidePopover();
+  }
+
+  protected openChanged() {
+    // this is from FAST tooltip: if ((this as FASTElement).$fastController.isConnected) {
+    if ((this as FASTElement).$fastController.isConnected) {
+      if (this.open) {
+        this.showPopover();
+      } else {
+        this.hidePopover();
+      }
+
+      Updates.enqueue(() => {
+        if (this.open) {
+          this.updatePosition();
+        }
+      });
+    }
+  }
+
+  private isOpen: boolean = false;
```

Comment:
Can you elaborate on the need for what seems to be either a dirty state or an internal private state?